packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -0,0 +1,64 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType } from 'echarts';
+import { ECHARTS_GESTURE } from '../../../common/constants';
+import { DEFAULT_DATA_ZOOM } from '../eChartsConstants';
+import { ViewportInMs } from '../types';
+import { Viewport } from '@iot-app-kit/core';
+
+// known bug with zooming into live data https://github.com/apache/echarts/issues/11679
+// this function will prevent the data from moving if we are zoomed in
+const onDataZoomEvent = (chart: EChartsType, setViewport: (viewport: Viewport, lastUpdatedBy?: string) => void) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  const currentDataZoom = chart.getOption().dataZoom[0];
+
+  const newDataZoom = {};
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.start > 0 ? (newDataZoom.startValue = currentDataZoom.startValue) : (newDataZoom.start = 0);
```

Comment:
`newDataZoom.startValue = currentDataZoom.start > 0 ? currentDataZoom.startValue : 0;`

Reply:
nope, we have to set either `startValue` or just `start` depending on what the current dataZoom is

Reply:
sure, i am not questioning the logic... when using tertiary op, it should return a value rather than assigning the value ? also, I am guessing you have ts-ignore for this ? ts-ignore should almost never be used

Reply:
ended up not needing this at all! less code wooo